src/variant.rs

```diff
@@ -143,6 +208,17 @@ impl Variant {
         }
     }
 
+    /// Creates a new GVariant array from children.
+    pub fn new_array<T: StaticVariantType>(children: &[Variant]) -> Self {
```

Comment:
Some kind of `append_to_array()` API could also be useful

Reply:
That would make `impl ToVariant for Vec<T>` a bit less wasteful (no need for an intermediate vec)

Reply:
A `Variant` is supposed to be non-mutable so there is no `_append` possible. If we want this, we need bindings for `VariantBuilder`. Hopefully basic bindinds for that are not hard.

Reply:
Ah I see. Something for another PR then unless you want to do it now :)

Reply:
Another PR seems good. I want this to be done ASAP. :) Let's create an issue for that and assign it to me.

Reply:
Created https://github.com/gtk-rs/glib/issues/676 for this. I think the PR is now good enough to be merged.